### PR TITLE
refactor - migrate simple components to stateless functional components [ready]

### PR DESCRIPTION
**Pull request by hannahstwinter**
Another PR migrating simple components to stateless functional components. There will probably be one more though it will only have two or so components and I'm stalling because the effects are trickling through to other spec files. 🙊

### CODE REVIEW

File: src/FormFieldValidationBlock.jsx [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-      </span>
-    ) : null;
-  }
-}
+const FormFieldValidationBlock = (props) => {
+  return props.value ? (
+    <span className="rs-validation-block">
+      <i className="rs-validation-indicator"></i> { props.value }
+    </span>
+  ) : <noscript />;
```
Comment (willcumbie) [inline src/FormFieldValidationBlock.jsx]: Why noscript rather than null?

File: test/FormFieldSpec.jsx [hunk 2]
```diff
@@ -1,72 +1,71 @@
...
 
-      message = TestUtils.findRenderedDOMComponentWithClass(formField, 'rs-validation-block');
-      expect(message.getDOMNode().textContent).toBe(' Test success message');
+      expect(validationBlock.props.value).toBe('Test success message');
     });
   });
 
-  it('adds the help message', () => {
-    let message;
+  // it('adds the help message', () => {
```
Comment (willcumbie) [inline test/FormFieldSpec.jsx]: Uncomment these.

Comment (hannahstwinter) [inline test/FormFieldSpec.jsx]: wherps

File: src/FormFieldValidationBlock.jsx (hunk 1 above)
Comment (hannahstwinter) [inline src/FormFieldValidationBlock.jsx]: Note section on https://facebook.github.io/react/docs/reusable-components.html#stateless-functions - "In React v0.14, stateless functional components were not permitted to return null or false (a workaround is to return a <noscript /> instead). This was fixed in React v15...". I think we're using 0.14.8?

**Comment by willcumbie** [comment]
LGTM

File: src/FormFieldValidationBlock.jsx (hunk 1 above)
Comment (hannahstwinter) [inline src/FormFieldValidationBlock.jsx]: lol - it ate my noscript tag

**Comment by j00k** [comment]
:ok_hand:

**Comment by adampickeral** [comment]
:+1: 

for future PR's, commit messages should look a little more like this:
```
refactor(stateless components): migrate components to stateless components
// anything else needed to explain the change in more detail
```
https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format

**Comment by hannahstwinter** [comment]
@adampickeral I rebased and fixed it. I meant to do that before now - thanks for reminding me! I think I forgot to do this on the earlier comparable PR.

**Pull request merged**